src/main/java/com/coreos/jetcd/exception/AuthFailedException.java

```diff
@@ -4,7 +4,7 @@
  * Signals that an error occurred while attempting to auth, this
  * may caused by wrong username or password.
  */
-public class AuthFailedException extends Exception {
+public class AuthFailedException extends RuntimeException {
 
   public AuthFailedException(String cause, Throwable throwable) {
```

Comment:
no public constructor.
 `AuthFailedException(String cause, Throwable throwable) {}`

use EtcdExceptionFactory to create this exception.
```java
// jetcd/exception/EtcdExceptionFactory.java
AuthFailedException newAuthFailedException(String cause, Throwable throwable) {
return new AuthFailedException(cause, throwable );
}
```